Scripts/travis-build-test.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+set -x -o pipefail
+
+if [[ "$SWIFT_BUILD" == "true" ]]; then
+    swift build
+    exit 0
+fi
+
+# -jobs -- specify the number of concurrent jobs
+# `sysctl -n hw.ncpu` -- fetch number of 'logical' cores in macOS machine
+xcodebuild -jobs `sysctl -n hw.ncpu` test -project DVR.xcodeproj -scheme ${TRAVIS_XCODE_SCHEME} -sdk ${TRAVIS_XCODE_SDK} \
```

Comment:
Is this script taken from the Travis macOS config script? Is there a way we could shell out to that instead (so that we don't need to duplicate the code)?

Reply:
Hey, I'm not sure I understand. This is not a duplicate exactly of the script that was inlined int eh travis.yml previously. It's standard `xcodebuild` arguments, some of which consume environment variables which are set by Travis.

> Is there a way we could shell out to that instead?
The travis yml is calling this script, as opposed to inlining the `xcodebuild` command.

Am I understanding correctly?

Reply:
Ah, my question was more around "is there a version of this command that comes in the Travis container that we can use instead of rewriting it ourselves?"

That is, we can supply these Travis-specific env vars in the .travis.yml, and Travis will generate an `xcodebuild` command for us. I _believe_ there's a `build.sh` script that is on the machine that's running the build, but I think it might be too late by the time our build has started to use this.

I don't think this should be a major blocker, but if we can figure out how to use the Travis env vars to generate an `xcodebuild` command rather than rewriting the command ourselves, we can save ourselves some headaches in the future.

Reply:
That said, I can't seem to find any documentation about this in the Travis docs. We can skip over this for now, unless you want to do any more digging or reverse engineering 😛

Reply:
Ahh I see @eliperkins . I'm not too sure to be honest. I just googled around for a bundled build.sh for mac builds on travis but didn't find anything myself (it was a quick google to fair).

I'm not sure how different it is to delegate to a bash script, as you could see before my change the `xcodebuild` command was inlined right under `script:` in the yml file. The benefit of doing things this way is that we can have more fine grained control (i.e. if `SWIFT_BUILD` is set, then avoid running `xcodebuild`) and add more complexity to our build/test script. Of course we could have also done this inline in the yml file, but then it looks cluttered. It's just an approach I've taken on a lot of projects with travis, but I'm happy to revert and inline everything if you prefer. Just let me know.

The build is passing again, and I reverted the one env variable for the platform specification.

Reply:
All good with me! Let's keep this as it is in this PR.

My main concern here was duplicating the usage of the Travis-specific variables. If Travis is going to dictate the API for variables like `TRAVIS_XCODE_SCHEME` and such, it would be good to ensure that we use their build command rather than duplicate it ourself.

Given that we can't seem to find how they use these variables beyond what we see in the output, I think it's fine to just roll our own. 👍

Reply:
cool. i can investigate more as well and try to de-dup in my follow-up which i'm interested in #80 

always happy to improve CI setup